chia/full_node/full_node.py

```diff
@@ -1153,42 +1153,36 @@ async def respond_block(
             ] = await self.blockchain.pre_validate_blocks_multiprocessing([block], npc_results)
+            prevalidation_done = time.time()
             if pre_validation_results is None:
                 raise ValueError(f"Failed to validate block {header_hash} height {block.height}")
             if pre_validation_results[0].error is not None:
                 if Err(pre_validation_results[0].error) == Err.INVALID_PREV_BLOCK_HASH:
-                    added: ReceiveBlockResult = ReceiveBlockResult.DISCONNECTED_BLOCK
-                    error_code: Optional[Err] = Err.INVALID_PREV_BLOCK_HASH
-                    fork_height: Optional[uint32] = None
+                    self.log.info(f"Disconnected block {header_hash} at height {block.height}")
+                    return None
                 else:
                     raise ValueError(
                         f"Failed to validate block {header_hash} height "
                         f"{block.height}: {Err(pre_validation_results[0].error).name}"
                     )
-            else:
-                result_to_validate = (
-                    pre_validation_results[0] if pre_validation_result is None else pre_validation_result
-                )
-                assert result_to_validate.required_iters == pre_validation_results[0].required_iters
-                added, error_code, fork_height = await self.blockchain.receive_block(block, result_to_validate, None)
-                if (
-                    self.full_node_store.previous_generator is not None
-                    and fork_height is not None
-                    and fork_height < self.full_node_store.previous_generator.block_height
-                ):
-                    self.full_node_store.previous_generator = None
-            validation_time = time.time() - validation_start
+
+            result_to_validate = pre_validation_results[0] if pre_validation_result is None else pre_validation_result
+            assert result_to_validate.required_iters == pre_validation_results[0].required_iters
+            added, error_code, fork_height = await self.blockchain.receive_block(block, result_to_validate, None)
+            if (
+                self.full_node_store.previous_generator is not None
+                and fork_height is not None
+                and fork_height < self.full_node_store.previous_generator.block_height
+            ):
+                self.full_node_store.previous_generator = None
+            validation_time = time.time()
 
             if added == ReceiveBlockResult.ALREADY_HAVE_BLOCK:
                 return None
             elif added == ReceiveBlockResult.INVALID_BLOCK:
                 assert error_code is not None
                 self.log.error(f"Block {header_hash} at height {block.height} is invalid with code {error_code}.")
                 raise ConsensusError(error_code, header_hash)
-
-            elif added == ReceiveBlockResult.DISCONNECTED_BLOCK:
```

Comment:
I just realized that this block has to be left here, since `self.blockchain.receive_block()` may also fail with this error code.

Reply:
there doesn't appear to be a test exercising this case though